src/engine/Settings.cs

```diff
@@ -26,85 +74,85 @@ private Settings()
     /// <summary>
     ///   Sets whether the game window is in fullscreen mode
     /// </summary>
-    public bool FullScreen { get; set; } = true;
+    public SettingValue<bool> FullScreen { get; set; }
```

Comment:
I wonder if it would be possible to have the default values stay here.
Does something like:
```
public SettingValue<bool> FullScreen { get; set; } = new SettingsValue<bool>(true);
```
compile?

Reply:
It should, I just wasn't sure if it was any cleaner since it's basically just moving the settings from the 'list' in the constructor to each property. It also doesn't work if I stop using automatic properties.

Reply:
I like to have the default values directly in the property definitions instead of in the constructors. As constructors aren't used in classes that don't need special behaviour on construction in the thrive codebase.